pkg/storage/azurecdn/storage.go

```diff
@@ -0,0 +1,87 @@
+package azurecdn
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/Azure/azure-storage-blob-go/2017-07-29/azblob"
+)
+
+// Storage implements (github.com/gomods/athens/pkg/storage).Saver and
+// also provides a function to fetch the location of a module
+type Storage struct {
+	accountURL *url.URL
+	cred       azblob.Credential
+}
+
+// New creates a new azure CDN saver
+func New(accountName, accountKey string) (*Storage, error) {
+	u, err := url.Parse(fmt.Sprintf("https://%s.blob.core.windows.net", accountName))
+	if err != nil {
+		return nil, err
+	}
+	cred := azblob.NewSharedKeyCredential(accountName, accountKey)
+	return &Storage{accountURL: u, cred: cred}, nil
+}
+
+// BaseURL returns the base URL that stores all modules. It can be used
+// in the "meta" tag redirect response to vgo.
+//
+// For example:
+//
+//	<meta name="go-import" content="gomods.com/athens mod BaseURL()">
+func (s Storage) BaseURL() *url.URL {
+	return s.accountURL
+}
+
+// Save implements the (github.com/gomods/athens/pkg/storage).Saver interface.
+func (s *Storage) Save(module, version string, mod, zip, info []byte) error {
+	ctx := context.Background()
```

Comment:
I see you've deleted the `Saver` interface, so I presume you're not yet bound to this method header. I'd update this to take a context as the first parameter, as so:

``` Go
func (s *Storage) Save(ctx context.Context, module, version string, mod, zip, info []byte) error {
```

I've found this pattern to be quite helpful, and adopting it later is a huge PITA.

Reply:
saver is from CDN package

Reply:
+1 on taking a `context.Context`. we'd have to change [this Saver interface](https://github.com/gomods/athens/blob/master/pkg/storage/saver.go) though, so I think we should do that in a later PR

Reply:
@marstr ref: https://github.com/gomods/athens/issues/174